treecorr/catalog_formats.py

```diff
@@ -0,0 +1,368 @@
+# Copyright (c) 2003-2020 by Mike Jarvis
+#
+# TreeCorr is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+This is a set of helper classes that read data for treecorr.Catalog objects.
+They have a bit of a mixed bag of functions, not intended to be complete,
+just to cover the needs of that class.
+
+HDF and FITS files differ in that the former can have different length columns
+in the same data group, unlike fits HDUs, which have columns of all the same length.
+Where possible we check for that here, and a user would have to be fairly determined to
+trigger this.
+
+The other difference is that fitsio reads a structured array, whereas h5py will
+read a dictionary of arrays. This does not cause problems here, since both are
+indexed by string, but may prevent usage elsewhere. If so we could convert them to
+both provide dicts.
+"""
+import numpy as np
+from treecorr.config import get_from_list
+import warnings
+
+class FitsReader:
+    """Reader interface for FITS files.
+    Uses fitsio to read columns, etc.
+    """
+    def __init__(self, file_name):
+        """Open a file
+
+        Parameters
+        ----------
+        file_name: str
+        """
+        import fitsio
+        # packaging is used by pip so is installed basically everywhere
+        from packaging.version import parse
+        self.file = fitsio.FITS(file_name, 'r')
+
+        # record file name to make error messages more useful
+        self.file_name = file_name
+
+        # There is a bug in earlier fitsio versions that prevents
+        # slicing
+        self.can_slice = parse(fitsio.__version__) > parse('1.0.6')
```

Comment:
Okay - doing it alphabetically seems to work for most cases I can construct, at least for comparing to '1.0.6'.  I'll change it.

Reply:
It's probably most relevant for numpy that seems to be going on with many past 1.9, where lexical comparison would break down.  Despite API changes at various versions, they seem weirdly reticent to bump to 2.x.